plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -77,11 +79,16 @@ public function enqueue_scripts() {
 		}
 		$post_type_object     = get_post_type_object( 'product' );
 		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => self::EDITOR_CONTEXT_NAME ) );
+		$template_registry 	  = wc_get_container()->get( BlockTemplateRegistry::class );
 
 		$editor_settings = array();
 		if ( ! empty( $post_type_object->template ) ) {
 			$editor_settings['template']                 = $post_type_object->template;
 			$editor_settings['templateLock']             = ! empty( $post_type_object->template_lock ) ? $post_type_object->template_lock : false;
+			$editor_settings['templates']     = array(
+				'product' 			=> $post_type_object->template,
+				'product_variation' => $template_registry->get_registered( 'product-variation' )->get_formatted_template()
+			);
```

Comment:
As we mentioned, good enough for now until we can get the templates from the REST API.

Reply:
Yup, it does kinda suck this gets added to every WCAdmin page 😢

Reply:
Yep, but we can do some logic here to recognize the type of product we are dealing with for the current request and only load the templates depending on that right?

Reply:
I don't think so, because once in the product editor the routing is done client-side, so we won't hit this again.

Reply:
> I don't think so, because once in the product editor the routing is done client-side, so we won't hit this again.

Right!